test/libcommonserver/io/testchecksetgetrights.cpp

```diff
@@ -0,0 +1,298 @@
+                rightsSetFromInt(baseRigths, rightsSet);
+
+                bool result = IoHelper::setRights(filepath, rightsSet.read, rightsSet.write, rightsSet.execute, ioError);
+                result &= ioError == IoErrorSuccess;
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to set base rights */);
+                }
+
+                result = IoHelper::getRights(filepath, isReadable, isWritable, isExecutable, exists);
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to get base rights */);
+                }
+
+                if (!(exists && isReadable == rightsSet.read && isWritable == rightsSet.write &&
+                      isExecutable == rightsSet.execute)) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Set base rights mismatch  with get base rights */);
+                }
+
+                rightsSetFromInt(targetRigths, rightsSet);
+
+                result = IoHelper::setRights(filepath, rightsSet.read, rightsSet.write, rightsSet.execute, ioError);
+                result &= ioError == IoErrorSuccess;
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to set target rights */);
+                }
+
+                result = IoHelper::getRights(filepath, isReadable, isWritable, isExecutable, exists);
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to get target rights */);
+                }
+
+                if (!(exists && isReadable == rightsSet.read && isWritable == rightsSet.write &&
+                      isExecutable == rightsSet.execute)) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Set target rights mismatch with get target rights */);
+                }
+            }
+        }
+
+        // Restore the rights
+        IoHelper::setRights(filepath, true, true, true, ioError);
+    }
+
+    // Check permission is not recursive on folder
+    {
+        const SyncPath path = temporaryDirectory.path / "testCheckSetAndGetRights";
+        const SyncPath subFolderPath = path / "subFolder";
+        const SyncPath subFilePath = path / "subFile.txt";
+
+
+        IoError ioError = IoErrorUnknown;
+        CPPUNIT_ASSERT(IoHelper::createDirectory(path, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
+
+        CPPUNIT_ASSERT(IoHelper::createDirectory(subFolderPath, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
+
+        std::ofstream file(subFilePath.string());
```

Comment:
Is the call to `.string()` required?

Reply:
After verification it is not. See commit: https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/dd6b5a0af9be3ce6d4dcfee81698ea2a2da085be.